tasks/insights-client.yml

```diff
@@ -38,6 +38,137 @@
     insertafter: "#auto_update"
     line: auto_update={{ rhc_insights.autoupdate | d(true) | bool }}
 
+- name: Configure authmethod
+  when:
+    - rhc_insights.authmethod is defined
+    - not rhc_insights.authmethod is none
+    - rhc_insights.authmethod != ""
+    - rhc_insights.authmethod != __rhc_state_absent
+    - rhc_insights.authmethod != omit
+  lineinfile:
+    path: "{{ __rhc_insights_conf }}"
+    regexp: "^authmethod="
+    state: present
+    line: authmethod={{ rhc_insights.authmethod }}
+  check_mode: true
+
+- name: Configure username for authmethod BASIC
+  when:
+    - rhc_insights.authmethod = "BASIC"
+    - rhc_insights.authmethod != __rhc_state_absent
+    - rhc_insights.username is defined
+    - not rhc_insights.username is none
+    - rhc_insights.username != ""
+    - rhc_insights.username != omit
+  lineinfile:
+    path: "{{ __rhc_insights_conf }}"
+    regexp: "^username="
+    state: present
+    line: username={{ rhc_insights.username }}
+  check_mode: true
+
+- name: Configure password for authmethod BASIC
+  when:
+    - rhc_insights.authmethod == "BASIC"
+    - rhc_insights.authmethod != __rhc_state_absent
+    - rhc_insights.password is defined
+    - not rhc_insights.password is none
+    - rhc_insights.password != ""
+    - rhc_insights.password != omit
+  lineinfile:
+    path: "{{ __rhc_insights_conf }}"
+    regexp: "^password="
+    state: present
+    line: password={{ rhc_insights.password }}
+  check_mode: true
```

Comment:
This will likely need `no_log: true` in order to prevent the cleartext password from being leaked to the ansible log.

Reply:
In fact - every task that can display the contents of `rhc_insights` will now need `no_log: true` in order to prevent leaking the cleartext `password`

To verify - run the playbook with `-vv` - then `grep` the logs/output for the given password.

Reply:
I see your point. I'm going to move authentication related variables to another dictionary. Then setting ˋno_log: trueˋ for all tasks accessing ˋrhc_insightsˋ won't be necessary anymore.

Thank you for your comments. :-)